src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/Worker.java

```diff
@@ -520,7 +518,7 @@ public Worker(
             IMetricsFactory metricsFactory, long taskBackoffTimeMillis, long failoverTimeMillis,
             boolean skipShardSyncAtWorkerInitializationIfLeasesExist, ShardPrioritization shardPrioritization,
             Optional<Integer> retryGetRecordsInSeconds, Optional<Integer> maxGetRecordsThreadPool, WorkerStateChangeListener workerStateChangeListener,
-            LeaseCleanupValidator leaseCleanupValidator) {
+            ShardSyncer shardSyncer) {
```

Comment:
Would chain this instead of changing it all together.

Reply:
LeaseCleanupValidator is used for initialization of ShardSyncer. Passing it separately doesn't make sense if we are passing in ShardSyncer. It's anyway a package-private constructor so there'll be no breaking changes for customers.